src/ExactDiagonalization/basis_bfs.jl

```diff
@@ -0,0 +1,365 @@
+const SubVector{T} = SubArray{T,1,Vector{T},Tuple{UnitRange{Int64}},true}
+"""
+    UniformSplit(array::Vector, min_chunk_size, max_chunks)
+
+Split the array into at most `max_chunks` subarrays, with each at least `min_chunk_size` long.
+
+```jldoctest
+julia> using Rimu.ExactDiagonalization: UniformSplit
+
+julia> UniformSplit(collect(1:10), 3, 3)
+3-element UniformSplit{Int64}:
+ [1, 2, 3, 4]
+ [5, 6, 7, 8]
+ [9, 10]
+
+julia> UniformSplit(collect(1:10), 5, 3)
+2-element UniformSplit{Int64}:
+ [1, 2, 3, 4, 5]
+ [6, 7, 8, 9, 10]
+```
+"""
+struct UniformSplit{T} <: AbstractVector{SubVector{T}}
+    array::Vector{T}
+    n_chunks::Int
+    chunk_size::Int
+
+    function UniformSplit(array::Vector{T}, min_chunk_size, max_chunks) where {T}
+        chunk_size = max(min_chunk_size, cld(length(array), max_chunks))
+        n_chunks = cld(length(array), chunk_size)
+        return new{T}(array, n_chunks, chunk_size)
+    end
+end
+
+Base.size(split::UniformSplit) = (split.n_chunks,)
+
+function Base.getindex(split::UniformSplit, i)
+    index_start = (i - 1) * split.chunk_size + 1
+    if i == split.n_chunks
+        index_end = length(split.array)
+    elseif 0 < i < split.n_chunks
+        index_end = i * split.chunk_size
+    else
+        throw(BoundsError(split, i))
+    end
+    return view(split.array, index_start:index_end)
+end
+
+"""
+    struct BasisBuilder
+
+Used with [`basis_bfs`](@ref) to build a basis from an operator.
```

Comment:
```suggestion
    bb! = BasisBuilder{A::Type}(; col_hint=0)
    bb!(operator::AbstractOperator, pairs, seen)

Functor used with [`basis_bfs`](@ref) to build a basis of addresses of type `A` from an operator.
It contains a set of addresses (`bb!.frontier`) that is collected from the `offdiagonals` of 
`operator` with all addresses contained the list of address-value pairs `pairs` that are not element of `seen`.
```
Please check if this is accurate. A bit more elaborate description what this does and how it is used would be helpful.